javaHapiValidatorLambda/src/main/java/software/amazon/fwoa/Handler.java

```diff
@@ -38,4 +59,51 @@ public ValidatorResponse handleRequest(String event, Context context) {
         return validate;
 
     }
+
+    private List<String> getBucketObjects(String bucketName) {
+        try {
+            List<String> objectKeys = new ArrayList<String>();
+            ListObjectsV2Request req = new ListObjectsV2Request().withBucketName(bucketName);
+            ListObjectsV2Result result;
+            do {
+                result = s3.listObjectsV2(req);
+
+                for (S3ObjectSummary objectSummary : result.getObjectSummaries()) {
+                    objectKeys.add(objectSummary.getKey());
+                }
+                // If there are more than maxKeys keys in the bucket, get a continuation token
+                // and list the next objects.
+                String token = result.getNextContinuationToken();
+                req.setContinuationToken(token);
+            } while (result.isTruncated());
+            log.info("found " + objectKeys.size() + " keys");
+            return objectKeys;
+        } catch (Exception e) {
+            throw new Error(e);
+        }
+    }
+
+    private Map<String, List<IGObject>> downloadObjects(List<String> keys, String bucketName) {
+        // keep track of .index.json objects
+        List<IGObject> indices = new ArrayList<IGObject>();
+        List<IGObject> resources = new ArrayList<IGObject>();
+        for (String key : keys) {
+            try (S3ObjectInputStream s3Object = s3.getObject(bucketName, key).getObjectContent()) {
+                IGObject bucketObj = new Utils.IGObject(key, IOUtils.toString(s3Object));
```

Comment:
So the validator do not need to care how many IGs are included, if there're two IGs to load, there'll be two .index.json file and related resource files do not need to be differentiated between two IGs?

Reply:
Yup each IG is placed into its own folder and will be indexed by the .index.json file.